ext/gd/libgd/gd_avif.c

```diff
@@ -342,6 +342,14 @@ gdImagePtr gdImageCreateFromAvifCtx (gdIOCtx *ctx)
 
 	decoder = avifDecoderCreate();
 
+	// Check if libavif version is >= 0.9.1
+#if AVIF_VERSION_MAJOR > 0 || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR > 9) || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR == 9 && AVIF_VERSION_PATCH >= 1)
+	// If so, allow the PixelInformationProperty ('pixi') to be missing in AV1 image
+	// items. libheif v1.11.0 or older does not add the 'pixi' item property to
+	// AV1 image items. (This issue has been corrected in libheif v1.12.0.)
```

Comment:
Is it possible to link with `libheif` instead of `libavif`?

Reply:
got a test failure when `libavif` is build without `libaom` https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/23479

Reply:
Is it due to just some minor variation of pixels? In that case we probably should use [similarity.inc](https://github.com/php/php-src/blob/master/ext/gd/tests/similarity.inc) for the comparison.

Reply:
Did you use a different encoder than `aom`? If so, which did you use?

For libgd, we decided to use `libaom`.

About `libheif` - this library can certainly create AVIF images as well! For various reasons, we decided to use `libavif` in gd. So I ultimately optimized my logic for libavif. To use `libheif` instead, you'd need to rewrite `gd_avif.c`.

Reply:
@morsssss thank you! I [used to build](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/23479/diffs) `libavif` with `aom` and [applied](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/22236/diffs) your patch - tests now passed on 0.9.2

Reply:
Meantime CI shows failures only on `armv7` and `armhf`

```
TEST 6112/15000 [ext/gd/tests/imagecreatefromstring_avif.phpt]
========DIFF========
[1;32m002+ Frame size limit reduced from 268435456 to 67108864.[0m
     int(250)
     int(375)
[1;32m006+ Frame size limit reduced from 268435456 to 67108864.[0m
     int(480)
     int(270)
========DONE========
[1;31mFAIL[0m imagecreatefromstring() - AVIF format [ext/gd/tests/imagecreatefromstring_avif.phpt] 
```
and
```
TEST 5930/15000 [ext/gd/tests/avif_decode_encode.phpt]
========DIFF========
[1;32m001+ Decoding AVIF image: Frame size limit reduced from 268435456 to 67108864.[0m
[1;32m002+ Frame size limit reduced from 268435456 to 67108864.[0m
[1;32m003+ ok[0m
[1;31m001- Decoding AVIF image: ok[0m
     Default AVIF encoding: ok
     Encoding AVIF at quality 70: ok
     Encoding AVIF at quality 70 with speed 5: ok
--
     Encoding AVIF with illegal quality: ok
     Encoding AVIF with illegal speed: ok
     Encoding AVIF losslessly... ok
[1;32m012+ Frame size limit reduced from 268435456 to 67108864.[0m
     How many pixels are different in the two images? 0
========DONE========
[1;31mFAIL[0m avif decoding/encoding tests [ext/gd/tests/avif_decode_encode.phpt] 
```

Reply:
Would be great to improve pipelines to add `libavif-dev` and run tests using it `--with-avif`

Reply:
Both the PR and https://github.com/php/php-src/commit/bb9ef2bedb378fb0f5271b0771bd0d9bba4a8483 still shows failures on arm (guess it caused by 32-bits)

Reply:
This test passes on 32bit Windows with libavif 0.9.0; so either it's only an issue with newer libavif, or not particularly 32bit related at all.

Reply:
Still fails on armv7 and armhf using beta2
```
TEST 6120/15019 [ext/gd/tests/imagecreatefromstring_avif.phpt]
========DIFF========
002+ Frame size limit reduced from 268435456 to 67108864.
     int(250)
     int(375)
006+ Frame size limit reduced from 268435456 to 67108864.
     int(480)
     int(270)
========DONE========
FAIL imagecreatefromstring() - AVIF format [ext/gd/tests/imagecreatefromstring_avif.phpt]

TEST 5938/15019 [ext/gd/tests/avif_decode_encode.phpt]
========DIFF========
001+ Decoding AVIF image: Frame size limit reduced from 268435456 to 67108864.
002+ Frame size limit reduced from 268435456 to 67108864.
003+ ok
001- Decoding AVIF image: ok
     Default AVIF encoding: ok
     Encoding AVIF at quality 70: ok
     Encoding AVIF at quality 70 with speed 5: ok
--
     Encoding AVIF with illegal quality: ok
     Encoding AVIF with illegal speed: ok
     Encoding AVIF losslessly... ok
012+ Frame size limit reduced from 268435456 to 67108864.
     How many pixels are different in the two images? 0
========DONE========
FAIL avif decoding/encoding tests [ext/gd/tests/avif_decode_encode.phpt]
```

Reply:
@andypost , please see the thread below.

Reply:
@morsssss FYI using patch to build libavif https://github.com/AOMediaCodec/libavif/commit/1556f21fb54e24d3612a0782fa0e5c8360bccbb9 allowed to pass tests)